tests/test_url_handler.py

```diff
@@ -0,0 +1,700 @@
+import re
+import json
+from string import digits, uppercase, capwords
+
+from django import http
+from django.conf import settings
+
+import pytest
+
+from nomination import url_handler, models
+import factories
+
+
+pytestmark = pytest.mark.django_db
+alnum_list = sorted(digits + uppercase)
+
+
+def test_alphabetical_browse():
+    surts = {
+        'A': ('http://(org,alarm,)', 'http://(org,a'),
+        'C': ('http://(org,charlie,)', 'http://(org,c'),
+        '1': ('http://(org,123,)', 'http://(org,1')
+    }
+    project = factories.ProjectFactory()
+    # Create the surts we're expecting to see represented in the returned dict.
+    [factories.SURTFactory(url_project=project, value=surts[key][0]) for key in iter(surts)]
+    expected = {'org': [(x, surts[x][1] if surts.get(x) else None) for x in alnum_list]}
+    # Create another unrelated SURT to make sure we aren't grabbing everything.
+    factories.SURTFactory()
+    returned = url_handler.alphabetical_browse(project)
+
+    assert returned == expected
+
+
+@pytest.mark.parametrize('surt, expected', [
+    ('http://(,)', {}),
+    ('http://(org,)', {'org': [(x, None) for x in alnum_list]})
+])
+def test_alphabetical_browse_domains_not_found(surt, expected):
+    project = factories.ProjectFactory()
+    factories.SURTFactory(url_project=project, value=surt)
+    returned = url_handler.alphabetical_browse(project)
+
+    assert returned == expected
+
+
+def test_alphabetical_browse_fake_project():
+    project = 'not a real project'
+    with pytest.raises(http.Http404):
+        url_handler.alphabetical_browse(project)
+
+
+def test_get_metadata():
+    project = factories.ProjectFactory()
+    vals = [x.metadata for x in factories.MetadataValuesFactory.create_batch(3)]
+    metadata = [factories.ProjectMetadataFactory(project=project, metadata=x) for x in vals]
+    expected = [(x, [models.Metadata_Values.objects.get(metadata=x).value]) for x in metadata]
+    returned = url_handler.get_metadata(project)
+
+    for each in returned:
+        assert str(each) in str(expected)
+
+
+def test_get_metadata_with_valueset():
+    project = factories.ProjectFactory()
+    valset = factories.ValuesetFactory()
+    vals = [factories.ValuesetValuesFactory(valueset=valset).value for i in range(3)]
+    metadata = factories.MetadataFactory(value_sets=[valset])
+    factories.ProjectMetadataFactory(project=project, metadata=metadata)
+    returned = url_handler.get_metadata(project)
+
+    for _, value_iter in returned:
+        for value in value_iter:
+            assert value in vals
```

Comment:
I agree with what you are saying, but nothing else came to mind at the moment since the returned value is an iterator rather than a list in this case. I'll try creating a list from the iterator and then comparing that with the vals list.